Microsoft.Toolkit.Uwp.UI.Controls/CameraPreview/CameraPreview.cs

```diff
@@ -0,0 +1,171 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Uwp.Helpers.CameraHelper;
+using Windows.Graphics.Imaging;
+using Windows.Media;
+using Windows.Media.Capture.Frames;
+using Windows.Media.Core;
+using Windows.Media.Playback;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Camera Control to preview video. Can subscribe to video frames, software bitmap when they arrive.
+    /// </summary>
+    public sealed class CameraPreview : Control, IDisposable
+    {
+        private CameraHelper _cameraHelper;
+        private MediaPlayer _mediaPlayer;
+        private ComboBox _frameSourceGroupCombo;
+        private MediaPlayerElement _mediaPlayerElementControl;
+        private TextBlock _videoPreviewText;
+        private TextBlock _videoPreviewErrorTextBlock;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CameraPreview"/> class.
+        /// </summary>
+        public CameraPreview()
+        {
+            this.DefaultStyleKey = typeof(CameraPreview);
+        }
+
+        /// <summary>
+        /// Event raised when a new video frame arrives.
+        /// </summary>
+        public event EventHandler<VideoFrame> VideoFrameArrived;
+
+        /// <summary>
+        /// Event raised when a new software bitmap arrives.
+        /// </summary>
+        public event EventHandler<SoftwareBitmap> SoftwareBitmapArrived;
```

Comment:
as said previously, I would get rid of this one and keep only the other one

Reply:
Agree, there should be one event

Reply:
you can get a SoftwareBitmap from a (VideoFrame)[https://docs.microsoft.com/en-us/uwp/api/Windows.Media.VideoFrame]: if its SoftwareBitmap is null, you can convert back and forth from Direct3DSurface to SoftwareBitmap easily. I'd would just send back VideoFrame, not SoftwareBitmap or VideoMediaFrame